### PR TITLE
FOGL-2461 Tests fixed

### PR DESCRIPTION
**Pull request by amarendra-dianomic**

### CODE REVIEW

File: test/test_coap_listen.py [hunk 1]
```diff
@@ -90,71 +90,24 @@ def test_plugin_init(mocker):
...
+    coap.plugin_start(config)
+    await asyncio.sleep(.3)
 
     # THEN
     assert coap.aiocoap_ctx is not None
-    assert 1 == log_info.call_count
+    assert 2 == log_info.call_count
     calls = [
         call('CoAP listener started on port {} with uri {}'.format(config['port']['value'], config['uri']['value']))]
     log_info.assert_has_calls(calls, any_order=True)
```
Comment (praveen-garg) [inline test/test_coap_listen.py:102]: can we verify the another call as well? I see now we have made `assert 2 == log_info.call_count`

File: test/test_coap_listen.py [hunk 2]
```diff
@@ -164,26 +117,29 @@ def test_plugin_reconfigure_else(mocker, unused_port):
...
+    await asyncio.sleep(.3)
 
     # THEN
-    assert 2 == log_info.call_count
-    calls = [call('CoAP listener started on port {} with uri sensor-values'.format(config['port']['value'])),
-             call('Stopping South CoAP plugin...')]
-    log_info.assert_has_calls(calls, any_order=True)
-    assert 0 == log_exception.call_count
+    assert new_config == new_handle
+    assert 3 == log_info.call_count
```
Comment (praveen-garg) [inline test/test_coap_listen.py:135]: We should assert shutdown init and start called in sequence

File: test/test_coap_listen.py [hunk 3]
```diff
@@ -197,17 +153,21 @@ def test_plugin_reconfigure_else(mocker, unused_port):
     log_info = mocker.patch.object(coap._LOGGER, "info")
 
     # WHEN
-    loop.call_soon(coap.plugin_start(config))
-    await asyncio.sleep(.5)  # required to allow ensure_future task to complete
+    coap.plugin_start(config)
+    await asyncio.sleep(.5)
```
Comment (praveen-garg) [inline test/test_coap_listen.py]: why this sleep is required. last sleep statement had a comment next to that.

File: test/test_coap_listen.py [hunk 4]
```diff
@@ -164,26 +117,29 @@ def test_plugin_reconfigure_else(mocker, unused_port):
...
+    new_config['uri']['value'] = new_config['uri']['default']
+    pstop = mocker.patch.object(coap, 'plugin_shutdown', return_value=True)
     log_info = mocker.patch.object(coap._LOGGER, "info")
 
     # WHEN
-    loop.call_soon(coap.plugin_start(config))
-    await asyncio.sleep(.5)  # required to allow ensure_future task to complete
-    coap._plugin_stop(config)
+    new_handle = coap.plugin_reconfigure(config, new_config)
+    await asyncio.sleep(.3)
```
Comment (praveen-garg) [inline test/test_coap_listen.py]: why this sleep is required. last sleep statement had a comment next to that.

**Review by praveen-garg — CHANGES_REQUESTED** [review]

File: test/test_coap_listen.py (hunk 1 above)
Comment (amarendra-dianomic) [reply test/test_coap_listen.py:102]: done

File: test/test_coap_listen.py (hunk 2 above)
Comment (amarendra-dianomic) [reply test/test_coap_listen.py:135]: done

File: test/test_coap_listen.py (hunk 4 above)
Comment (amarendra-dianomic) [reply test/test_coap_listen.py]: done

File: test/test_coap_listen.py (hunk 2 above)
Comment (praveen-garg) [reply test/test_coap_listen.py:135]: Add init patch call assertion as well; as we have for stop/shutdown

**Comment by praveen-garg** [comment]
We should remove comments like; these are sort of debug comment and should be avoided in prod mode. 

LOGGER.info("plugin_start called") 

Or its content must be similar to what we have in shutdown or reconfigure log statements. 

Also there are few inline comments in plugin reconfigure e.g. # plugin_init where the next line is a straight forward call itself as plugin_init. We should avoid such over statements

**Review by praveen-garg — APPROVED** [review]

File: test/test_coap_listen.py [hunk 5]
```diff
@@ -256,57 +230,25 @@ class TestCoapSouthIngest(object):
...
             "key": "80a43623-ebe5-40d6-8d80-3f892da9b3b4",
             "readings": "500"
         }"""
         with patch.object(coap._LOGGER, "exception") as log_exception:
-            with patch.object(Ingest, 'increment_discarded_readings', return_value=True) as ingest_discarded:
-                with patch.object(Ingest, 'is_available', return_value=True) as is_ingest_available:
-                    with pytest.raises(aiocoap.error.BadRequest) as excinfo:
-                        request = message.Message(payload=cbor2.dumps(json.loads(data)), code=numbers.codes.Code.POST)
-                        r = await CoAPIngest.render_post(request)
+            with patch.object(async_ingest, 'ingest_callback') as ingest_add_readings:
```
Comment (praveen-garg) [inline test/test_coap_listen.py:237]: we should also have asserted ingest_add_readings call count 0 (i.e. never called)

File: python/foglamp/plugins/south/coap/coap.py [hunk 6]
```diff
@@ -108,8 +110,8 @@ def plugin_start(handle):
     Returns:
     Raises:
     """
-    global _task, loop
-    _LOGGER.info("plugin_start called")
+    global _task, loop, t
+    _LOGGER.debug("plugin_start called")
```
Comment (praveen-garg) [inline python/foglamp/plugins/south/coap/coap.py]: I was in favor to remove such comments and test patched calls

File: python/foglamp/plugins/south/coap/coap.py [hunk 7]
```diff
@@ -95,6 +96,7 @@ def plugin_init(config):
         handle: JSON object to be used in future calls to the plugin
     Raises:
     """
+    _LOGGER.debug("plugin_init called")
```
Comment (praveen-garg) [inline python/foglamp/plugins/south/coap/coap.py]: I was in favor to remove such comments and test patched calls. Instead of adding new

**Review by praveen-garg — DISMISSED** [review]
lgtm! We can remove debug logs and add 1 more assertion for test_render_post_reading_not_dict in FOGL-2461patch

File: python/foglamp/plugins/south/coap/coap.py (hunk 6 above)
Comment (amarendra-dianomic) [reply python/foglamp/plugins/south/coap/coap.py]: Such debug logs help in code coverage whereas method patching bypasses method body. Hence debug logs are preferred. Patching should be used only for those methods which play a secondary role in the method under testing, e.g. logger.

File: python/foglamp/plugins/south/coap/coap.py (hunk 7 above)
Comment (amarendra-dianomic) [reply python/foglamp/plugins/south/coap/coap.py]: Such debug logs help in code coverage whereas method patching bypasses method body. Hence debug logs are preferred. Patching should be used only for those methods which play a secondary role in the method under testing, e.g. logger.

File: test/test_coap_listen.py (hunk 2 above)
Comment (amarendra-dianomic) [reply test/test_coap_listen.py:135]: plugin_shutdown() patching has been removed.

**Review by praveen-garg — DISMISSED** [review]
We should not make logging a test coverage criteria; for coverage only unit under test matters and we can patch all other unit calls with expected (or no return value (as those are being tested as other units.
We can merge it.

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**